src/Common/src/Interop/Windows/Winsock/Interop.WSAGetOverlappedResult.cs

```diff
@@ -11,11 +11,11 @@ internal static partial class Interop
     internal static partial class Winsock
     {
         [DllImport(Interop.Libraries.Ws2_32, SetLastError = true)]
-        internal static extern bool WSAGetOverlappedResult(
-            [In] SafeCloseSocket socketHandle,
-            [In] SafeHandle overlapped,
-            [Out] out uint bytesTransferred,
-            [In] bool wait,
-            [Out] out SocketFlags socketFlags);
+        internal static unsafe extern bool WSAGetOverlappedResult(
+            SafeCloseSocket socketHandle,
+            SafeHandle overlapped,
+            uint* bytesTransferred,
```

Comment:
I think, in situations where passing `null` is strictly not allowed, we should use `out` or `ref` to correctly indicate the behavior here.

In situations where passing `null` is allowed, I think we have two scenarios:
* We don't have code that needs to pass in `null`
* We do have code that needs to pass in `null`

In the case of the former (we don't have code) then I think we should explicitly declare it as if the parameter was not optional (using out/ref instead) and the method declaration should be explicitly documented as such (just have a comment indicating 'this field is optional, but we have no cases where this is used today`).

In the case of the latter (we do have cases where we pass in `null`) we should declare it using pointers and make that invocation private. We should then expose 2 public overloads:
* One that declares the optional variable using out/ref
* One that does not expose the variable to the user at all

For the former, we just need to take the address of the parameter and pass it to the private invocation (it should already be pinned).

For the latter, we just need to pass in null